### ISSUE TITLE
Refactor Platform.sh deploy.py

### ISSUE THREAD

**Issue by ehmatthes** [ehmatthes/django-simple-deploy#286]
- [x] Refactor Platform.sh deploy.py, based on experience of refactoring fly and simple_deploy.py.
- [x] Use a more consistent style for docstrings. See [PEP 257](https://peps.python.org/pep-0257/). See also `_get_deployed_project_name()` as an [example](https://github.com/ehmatthes/django-simple-deploy/blob/main/simple_deploy/management/commands/fly_io/deploy.py#L376).
- [x] Run black.
- [ ] Run tests.
    - [x] local
    - [ ] e2e
- [x] Quick look at documentation
- [x] Finish simple refactoring.
- [x] Update CLI messages.
- [x] Merge.
- [x] Move bigger tasks to separate issue about extending support to multiple projects.

Resources
---

- [cli reference](https://docs.platform.sh/administration/cli/reference.html)

**Comment by ehmatthes** [comment]
Methods
---

- [ ] `__init__()`
- [ ] `deploy()`
- [ ] `_confirm_automate_all()`
- [ ] `_add_platformsh_settings()`
- [ ] `_get_platformsh_settings()`
- [ ] `_generate_platform_app_yaml()`
- [ ] `_add_requirements()`
- [ ] `_check_allowed_hosts()`
- [ ] `_make_platform_dir()`
- [ ] `_generate_services_yaml()`
- [ ] `_conclude_automate_all()`
- [ ] `_show_success_message()`
- [ ] `_confirm_preliminary()`
- [ ] `_validate_platform()`
- [ ] `_prep_automate_all()`
- [ ] `_validate_cli()`
- [ ] `_get_platformsh_project_name()`
- [ ] `_get_org_name()`
- [ ] `_confirm_use_org_name()`

**Comment by ehmatthes** [comment]
Working notes
---

- [ ] Group yaml generation steps together.
- [ ] Order of methods should roughly match what's listed in `deploy()`.

**Comment by ehmatthes** [comment]
Parse csv instead of default table
---

Output of `platform project:into --yes --format csv`:

```bash
$  platform project:info --yes --format csv  
Property,Value
id,<project-id>
created_at,2024-04-28T04:56:53+00:00
updated_at,2024-04-28T04:58:27+00:00
attributes,{  }
title,my_blog_project
description,
owner,<owner-hash>
namespace,platformsh
organization,<org-id>
default_branch,main
status,"code: provisioned
message: ok"
timezone,America/Denver
region,us-3.platform.sh
repository,"url: '<id>@git.us-3.platform.sh:<id>.git'
client_ssh_key: <redacted>
default_domain,
subscription,"<redacted>
plan: development
environments: 3
storage: 5120
included_users: 1
subscription_management_uri: 'https://console.platform.sh/-/users/<id>/...
restricted: false
suspended: false
user_licenses: 1"
...
```

**Comment by ehmatthes** [comment]
More notes
---

- [ ] Clean up all this path handling work. Define the path once at the beginning of the method. Report we're looking for that specific path.

**Comment by ehmatthes** [comment]
Closed by #309.
